packages/hydrogen/src/cache/server-fetch.ts

```diff
@@ -5,43 +5,43 @@ import {
   type DebugOptions,
 } from './run-with-cache.js';
 import type {WaitUntil} from '../types.js';
+import {parseJSON} from '../utils/parse-json.js';
 
-export type FetchCacheOptions = {
+export type FetchCacheOptions<T = any> = {
   cache?: CachingStrategy;
   cacheInstance?: Cache;
   cacheKey?: CacheKey;
-  shouldCacheResponse?: (body: any, response: Response) => boolean;
+  shouldCacheResponse?: (body: T, response: Response) => boolean;
```

Comment:
I thought we want to make this function as a required field

Reply:
I have this in the original post questions:

>fetch already checks for status errors (only caches 2xx responses) but there could still be situations where devs don't want to cache the result based on body (e.g. GraphQL APIs tend to return errors in body with a 200 status). For this, you can pass shouldCacheResponse parameter optionally. Do we want to make this a required param instead to force devs check the integrity of the response body?


The thing is, by creating the `withCache.fetch` counter part, I think this change is not required anymore. That said, we could still do it if we think devs are still going to shoot their feet. Alternatively, we could consider changing the `withCache.run` to support dynamic cache options, as discussed [here](https://github.com/Shopify/hydrogen/discussions/2558#discussioncomment-10732751).

Reply:
+1 for `required`